ui/src/pages/SignIn.js

```diff
@@ -1,48 +1,166 @@
 import React, { useState } from 'react';
-import { Button, Input, Header, Form, Label } from 'semantic-ui-react';
+import { Button, Input, Form, Icon } from 'semantic-ui-react';
+import mantaLogo from "../icons/manta.png";
+import dolphinLogo from "../icons/Square150x150Logo.png";
+import calamariLogo from "../icons/calamari.png";
+import newAccount from "../icons/new_account.png";
+import "../App.css";
+import HyperLinkButton from '../components/HyperLinkButton';
 
-const SignIn = ({ sendPassword, endInitialConnectionPhase }) => {
+const SignIn = ({
+  sendSelection,
+  getReceivingKeys,
+  receivingKey,
+  receivingKeyDisplay,
+  sendPassword,
+  endInitialConnectionPhase,
+  startRecover,
+  hideWindow
+}) => {
   const [password, setPassword] = useState('');
   const [passwordInvalid, setPasswordInvalid] = useState(null);
-  const [badPasswordTried, setBadPasswordTried] = useState(false);
+  const [loginSuccess, setLoginSuccess] = useState(false);
+  const [showCopyNotification, setShowCopyNotification] = useState(false);
+  const [loading, setLoading] = useState(false);
 
   const onClickSignIn = async () => {
+    setLoading(true);
+    await sendSelection("SignIn");
     const shouldRetry = await sendPassword(password);
+
     if (!shouldRetry) {
-      setPassword('');
-      console.log("[INFO]: End Initial Connection Phase");
+
+      await getReceivingKeys();
       await endInitialConnectionPhase();
+      setPassword('');
+      setLoginSuccess(true);
     } else {
       console.log("RETRY!");
       setPasswordInvalid(true);
-      setBadPasswordTried(true);
     }
+    setLoading(false);
   };
 
   const onChangePassword = password => {
-    setBadPasswordTried(false);
     setPassword(password)
     setPasswordInvalid(false)
   }
 
-  return (
-    <div>
-      <Header> Sign in </Header>
-      <Form.Field>
-        <Input
-          type="password"
-          placeholder="Password"
-          value={password}
-          onChange={(e) => onChangePassword(e.target.value)}
-          error={passwordInvalid}
+  const onClickCopyZkAddress = () => {
+    navigator.clipboard.writeText(receivingKey);
+
+    if (!showCopyNotification) {
+      setShowCopyNotification(true);
+      setTimeout(function () { setShowCopyNotification(false) }, 2000);
+    }
+
+  }
+
+  const onClickForgotPassword = async () => {
+    startRecover();
+  }
+
+  const onClickFinishSignIn = async () => {
+    await hideWindow();
+  }
+
+  return (<>
+    {!loginSuccess &&
```

Comment:
You probably want to factor out subcomponents here, one for `loginSuccess` and one for `!loginSuccess`. In general, conditional renders and especially nested conditional (here `!loginSuccess &&` and `loading ?`) are a signal that you want to spin off some components.